src/Microsoft.AspNet.Mvc/MvcServices.cs

```diff
@@ -37,9 +37,9 @@ public static IEnumerable<IServiceDescriptor> GetDefaultServices(IConfiguration
                 typeof(INestedProviderManagerAsync<>),
                 typeof(NestedProviderManagerAsync<>));
             yield return describe.Transient<MvcMarkerService, MvcMarkerService>();
-            yield return describe.Scoped(typeof(IScopedInstance<>), typeof(ScopedInstance<>));
-
-            // Core action discovery, filters and action execution.
+			yield return describe.Singleton(typeof(ITypeActivatorCache), typeof(DefaultTypeActivatorCache));
```

Comment:
Lifetime of `VirtualPathRazorPageFactory` is scoped now. Needs to be become singleton for caching to take effect.